### ISSUE TITLE
💅 Only removing unused imports is not working properly

### ISSUE THREAD

**Issue by redsuperbat** [biomejs/biome#3738]
### Environment information

```bash
CLI:
  Version:                      1.8.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "screen-256color"
  JS_RUNTIME_VERSION:           "v20.11.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.2.4"

Biome Configuration:
  Status:                       unset

Workspace:
  Open Documents:               0
```

### Rule name

lint/correctness/noUnusedImports

### Playground link

https://biomejs.dev/playground/?code=aQBtAHAAbwByAHQAIAB7ACAAdABlAHMAdAAgAH0AIABmAHIAbwBtACAAJwB0AGUAcwB0ACcAOwA%3D

### Expected result

Biome finds the error and with the `--error-on-warnings` flag it should properly fix the file. It does not fix the file.

As you can see from the command:

```sh
> npx @biomejs/biome lint --write --error-on-warnings --only=correctness/noUnusedImports ./services/CustomersService.ts
```

And the output:

```sh
./services/CustomersService.ts:2:21 lint/correctness/noUnusedImports  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━

  ⚠ This import is unused.

    1 │ import { type HttpClient, HttpError } from "./HttpClient";
  > 2 │ import type { Cart, CartItem } from "./models/Cart";
      │                     ^^^^^^^^
    3 │ import type { Customer } from "./models/Customer";
    4 │

  ℹ Unused imports might be the result of an incomplete refactoring.

  ℹ Safe fix: Remove the unused import.

    2 │ import·type·{·Cart,·CartItem·}·from·"./models/Cart";
      │                     ---------

Checked 1 file in 2ms. No fixes applied.
Found 1 warning.
lint ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Some warnings were emitted while running checks.
```

Biome did not write the fix even though it found an error.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Duplicate of https://github.com/biomejs/biome/issues/3470
